src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -27,13 +27,15 @@
 #include <app/CommandHandler.h>
 #include <app/ConcreteCommandPath.h>
 #include <app/server/CommissioningWindowManager.h>
+#include <app/server/EnhancedSetupFlowProvider.h>
 #include <app/server/Server.h>
 #include <app/util/attribute-storage.h>
 #include <lib/support/Span.h>
 #include <lib/support/logging/CHIPLogging.h>
 #include <platform/CHIPDeviceConfig.h>
 #include <platform/ConfigurationManager.h>
 #include <platform/DeviceControlServer.h>
+#include <system/SystemConfig.h>
```

Comment:
What causes this new header dependency? Curious why it shows up.

Reply:
```
/* Include a CHIP project-specific configuration file, if defined.
 *
 * An application or module that incorporates CHIP can define a project
 * configuration file to override standard CHIP configuration with
 * application-specific values.  The CHIPProjectConfig.h file is typically
 * located outside the CHIP source tree, alongside the source code for the
 * application.  The config file is included here to enable certain system-wide
 * configuration options, primarily related to logging and error reporting.
 */
#ifdef CHIP_PROJECT_CONFIG_INCLUDE
#include CHIP_PROJECT_CONFIG_INCLUDE
#endif

```

Reply:
SystemConfig.h includes CHIP_PROJECT_CONFIG_INCLUDE, which is set to `chip_project_config_include` which is set to `chip_project_config_include = "<CHIPProjectAppConfig.h>"`, which we need for the `
#define CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS` and `CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS_VERSION`

```
  if (chip_project_config_include != "") {
    defines += [ "CHIP_PROJECT_CONFIG_INCLUDE=${chip_project_config_include}" ]
  }
```